src/Microsoft.AspNet.OData.Shared/DataModificationExceptionType.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Builder;
+
+namespace Org.OData.Core.V1
+{
+    /// <summary>
+    /// Represents a Message Type
+    /// </summary>
+    public class MessageType
+    {
+        /// <summary>
+        /// Code of message
+        /// </summary>
+        public string Code { get; set; }
+
+        /// <summary>
+        /// Actual message
+        /// </summary>
+        public string Message { get; set; }
+
+        /// <summary>
+        /// Severity of message
+        /// </summary>
+        public string Severity { get; set; }
+
+        /// <summary>
+        /// Target of message
+        /// </summary>
+        public string Target { get; set; }
+
+        /// <summary>
+        /// Details of message
+        /// </summary>
+        public string Details { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception Type
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1012:AbstractTypesShouldNotHaveConstructors")]
+    public abstract class ExceptionType
+    {
+        /// <summary>
+        /// Instanciates a type of <see cref="ExceptionType"/>
+        /// </summary>
+        public ExceptionType()
+        {
+            ODataConventionModelBuilder builder = new ODataConventionModelBuilder();
+            builder.EntityType<DataModificationExceptionType>();
+      
+            builder.Namespace = typeof(DataModificationExceptionType).Namespace;
+        }
+
+        /// <summary>
+        /// Represents a MessageType
+        /// </summary>
+        public MessageType MessageType { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception for Data modification Operation.
+    /// </summary>    
+    public class DataModificationExceptionType : ExceptionType
+    {
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="DataModificationExceptionType"/> class.
+        /// </summary>
+        public DataModificationExceptionType(DataModificationOperationKind failedOperation)
+        {
+            this.failedOperation = failedOperation;
+        }
+
+        /// <summary>
+        /// Represents king of <see cref="DataModificationOperationKind"/> type of operation 
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1709:IdentifiersShouldBeCasedCorrectly", MessageId = "failed")]
+        public DataModificationOperationKind failedOperation { get; set; }
```

Comment:
>failedOperation [](start = 45, length = 15)

Are you sure use the camelcase for the public property name? 
It's same as for "responseCode"?

Reply:
In the vocabs its like that

---
In reply to: [604610652](https://github.com/OData/WebApi/pull/2455#discussion_r604610652) [](ancestors = 604610652)

Reply:
The casing of the annotation reflects the casing in Error payload, which is lower camel-case (unlike other term properties).

However, the public API for ODataError still uses upper-camel case for consistency.

We should do the same thing here -- the public properties should be upper camel case, even though we serialize in lower camel case.

---
In reply to: [604691364](https://github.com/OData/WebApi/pull/2455#discussion_r604691364) [](ancestors = 604691364,604610652)